daxa/process/erosita/_common.py

```diff
@@ -0,0 +1,445 @@
+                os.makedirs(stor_dir)
+    
+        # We also ensure that an overall directory for failed processing observations exists - this will give
+        #  observation directories which have no useful data in (i.e. they do not have a successful final
+        #  processing step) somewhere to be copied to (see daxa.process._cleanup._last_process).
+        # This is the overall path, there might not ever be anything in it, so we don't pre-make ObsID sub-directories
+        fail_proc_dir = obs_archive.get_failed_data_path(miss, None).format(oi='')[:-1]
+        if not os.path.exists(fail_proc_dir):
+            os.makedirs(fail_proc_dir)
+
+    return esass_in_docker
+
+def execute_cmd(cmd: str, esass_in_docker: bool, rel_id: str, miss_name: str, check_path: str,
+                extra_info: dict, timeout: float = None) -> Tuple[str, str, List[bool], str, str, dict]:
+    """
+    This is a simple function designed to execute eSASS commands either through Docker or the command line
+    for the processing and reduction of eROSITA mission data. It will collect the stdout and stderr values 
+    for each command and return them too for the process of logging. Finally, it checks that a specified 'final file' 
+    (or a set of 'final files') actually exists at the expected path, as a final check of the success of whatever 
+    process has been run.
+
+    :param str cmd: The command that should be executed in a bash shell.
+    :param Bool esass_in_docker: Set to True if eSASS is being used via Docker.
+    :param str rel_id: Whatever ID has been attached to the particular command (it could be an ObsID, or an ObsID
+        + instrument combination depending on the task).
+    :param str miss_name: The specific eROSITA mission name that this task belongs to.
+    :param str/list check_path: The path (or a list of paths) where a 'final file' (or final files) should exist, used
+        for the purposes of checking that it (they) exists.
+    :param dict extra_info: A dictionary which can contain extra information about the process or output that will
+        eventually be stored in the Archive.
+    :param float timeout: The length of time (in seconds) which the process is allowed to run for before being
+        killed. Default is None, which is supported as an input by communicate().
+    :return: The rel_id, a list of boolean flags indicating whether the final files exist, the std_out, and the
+        std_err. The final dictionary can contain extra information recorded by the processing function.
+    :rtype: Tuple[str, str, List[bool], str, str, dict]
+    """
+    # Either a single path or a list of paths can be passed to check - I make sure that the checking process only
+    #  ever has to deal with a list
+    if isinstance(check_path, str):
+        check_path = [check_path]
+
+    # eSASS is also released in a Docker container for Mac OS and Windows users, which is not yet supported in DAXA.
+    if esass_in_docker:
```

Comment:
yes it does, I just wanted to leave this in here, because whenever we do get esass in docker working, I will have to add to this indent, because the execute command function will have to change

Reply:
Ah right - very sensible